common/djangoapps/third_party_auth/views.py

```diff
@@ -152,3 +156,35 @@ def get(self, request, *args, **kwargs):
             return redirect(url)
         except ValueError:
             return HttpResponseNotFound()
+
+class SAMLProviderConfigViewSet(viewsets.ViewSet):
```

Comment:
Could you use ModelViewSet here and get all the CRUD actions? (I'm not super familiar with DRF Viewsets, but it looked like an interesting possiblility.

Reply:
yes I switched to ModelViewSet which basically works in the browser and via CURL just not able to get reverse to work yet

Reply:
so yeah this works on terminal

```
curl -X GET "http://localhost:18000/auth/saml/v0/providerconfig/1/"
```

as does this

```
curl -X GET "http://localhost:18000/auth/saml/v0/providerconfig/"
```